libtransmission/peer-mgr.cc

```diff
@@ -1213,6 +1213,8 @@ size_t tr_peerMgrAddPex(tr_torrent* tor, tr_peer_from from, tr_pex const* pex, s
             ++n_used;
         }
     }
+    // best estimate: we can't tell for sure if they are the same peers
+    s->stats.known_peer_from_count[from] = std::max(s->stats.known_peer_from_count[from], (uint16_t)n_used);
```

Comment:
You can refactor `tr_swarm::ensure_info_exists()` to return an `std::pair<tr_peer_info*, bool>`, where the `bool` means whether the peer is new to us. Then you can know for sure if "they are the same peers".

The current calculation *is* a best estimate, but it is still far from accurate.

Reply:
I don't think it would work: the `ensure_info_exists()` api is called from two places (lines 1248 and 1340) while we only collect stats from one of them.
But I could attempt to collect the stats from within `ensure_info_exists()`.

Reply:
The idea works, but it is not yet implemented properly right now, see the thread at https://github.com/transmission/transmission/pull/4900#discussion_r1377413682.